src/sgame/sg_admin.cpp

```diff
@@ -80,6 +80,35 @@ struct g_admin_command_t
 	char                   flag[ MAX_ADMIN_FLAG_LEN ];
 };
 
+struct g_admin_spec_t
+{
+	g_admin_spec_t     *next;
+	char               guid[ 33 ];
+	// -1: can not join until next game
+	// otherwise: expires - t : can not join until delta
+	int                expires;
+};
+
+struct g_admin_admin_t
+{
+	g_admin_admin_t      *next;
+
+	int                  level;
+	char                 guid[ 33 ];
+	char                 name[ MAX_NAME_LENGTH ];
+	char                 flags[ MAX_ADMIN_FLAGS ];
+	char                 pubkey[ RSA_STRING_LENGTH ];
+	char                 msg[ RSA_STRING_LENGTH ];
+	char                 msg2[ RSA_STRING_LENGTH ];
+	qtime_t              lastSeen;
+	int                  counter;
+};
+
+int G_admin_joindelay( g_admin_spec_t const* ptr )
```

Comment:
tbh, I do not really understand how the delay stuff works exactly, so it's a bit hard to document. In this case, it's only a setter, but "C-style", that is, the struct is defined in the cpp file, thus it's definition is not accessible from outside. Instead, functions which only exposes what is really used from the outside are exposed. This is comparable to a "private" in C++-style code.
The name might be badly chosen thought. It's only "inspired" from what I suppose the member accessed really is (`expires`) from what I understood of context.

This commit is really about having everything which does the admin auth stuff in a single place instead of spread into 3 cpp files.